src/utils/codapHelpers.ts

```diff
@@ -0,0 +1,386 @@
+           } else {
+               return Promise.resolve("Unknown attribute.");
+           }
+       }
+   });
+   return Promise.all(promises);
+}
+
+const hasDataset = async (name: string) => {
+   const result = await getDataContext(name);
+   return result.success === true;
+};
+
+const createMap = async (name: string, dimensions: IDimensions, center: ILatLong, zoom: number) => {
+   let map;
+
+   let componentListResult = await codapInterface.sendRequest({
+       action: "get",
+       resource: "componentList"
+   }) as IResult;
+
+   if (componentListResult && componentListResult.success) {
+       map = componentListResult.values.find(function (component: Record<string, string>) { return component.type==="map";});
+   }
+
+   if (!map) {
+       let result = await codapInterface.sendRequest({
+           action: "create", resource: "component", values: {
+               type: "map",
+               name,
+               dimensions,
+               dataContextName: name,
+               legendAttributeName: "isActive"
+           }
+       }) as IResult;
+       if (result.success) {
+           map = result.values;
+       }
+   }
+   if (map && center && (zoom != null)) {
+       return centerAndZoomMap(map.id, center, zoom);
+   } else {
+       return selectComponent(map.id);
+   }
+};
+
+const centerAndZoomMap = (mapName: string, center: ILatLong, zoom: number) => {
+   return new Promise<void>((resolve) => {
+       setTimeout(function () {
+           codapInterface.sendRequest({
+               action: "update",
+               resource: `component[${mapName}]`,
+               values: {
+                   center,
+                   zoom: 4
+               }
+           });
+           setTimeout(function () {
+               codapInterface.sendRequest({
+                   action: "update",
+                   resource: `component[${mapName}]`,
+                   values: {
+                       zoom
+                   }
+               });
+
+           }, 500);
+       }, 2000);
+       resolve();
+   });
+};
+
+const hasMap = async () => {
+   const componentsResult = await codapInterface.sendRequest({
+       action: "get",
+       resource: "componentList"
+   }) as IResult;
+   return componentsResult
+       && componentsResult.success
+       && componentsResult.values.find(function (component: IMapComponent) {
+           return component.type === "map";
+       });
+};
+
+const createStationsDataset = async (stations: IWeatherStation[]) => {
+  let result = await createDataContext(kStationsDatasetName);
+  if (!result.success) {
+      console.log(`Dataset, "${kStationsDatasetName}", creation failed`);
+      return;
+  }
+
+  result = await createNewCollection(kStationsDatasetName, kStationsCollectionName, kWeatherStationCollectionAttrs);
+  if (!result.success) {
+    console.log(`Collection, "${kStationsCollectionName}", creation failed`);
```

Comment:
Add a linting exception so we don't get warning

Reply:
Or make it a `console.warn` and configure the linter to allow warnings.